File: MVC Mode/src/TeacherView.java
import java.util.Locale;
import java.util.Map;

public class TeacherView {
    public void printTeacherDetails(String name, String subject, int gender, int code){
        String personalPronoun = judgePersonalPronoun(gender);
        String subjectPersonalPronoun = judgeSubjectPersonalPronoun(gender);

        System.out.println( subjectPersonalPronoun + " is a teacher, " + personalPronoun + " name is " + name + ".");
        System.out.println( subjectPersonalPronoun + " teaches subject is " + subject + ".");
        System.out.println( subjectPersonalPronoun + " employee number is " + code + ".");
    }

    public String judgePersonalPronoun(int gender){
        return gender == 1 ? "his" : "her";
    }

    public String judgeSubjectPersonalPronoun(int gender){
        return gender == 1 ? "he" : "she";
    }
}

File: MVCPattern/src/TeacherView.kt
class TeacherView {
    fun printTeacherDetails(name: String?, subject: String?, gender: Int, code: Int) {
        val personalPronoun = judgePersonalPronoun(gender)
        val subjectPersonalPronoun = judgeSubjectPersonalPronoun(gender)
        println("$subjectPersonalPronoun is a teacher, $personalPronoun name is $name.")
        println("$subjectPersonalPronoun teaches subject is $subject.")
        println("$subjectPersonalPronoun employee number is $code.")
    }

    fun judgePersonalPronoun(gender: Int): String {
        return if (gender == 1) "his" else "her"
    }

    fun judgeSubjectPersonalPronoun(gender: Int): String {
        return if (gender == 1) "he" else "she"
    }
}